src/appshell/view/dockwindow/dockpageview.cpp

```diff
@@ -56,6 +56,11 @@ QString DockPageView::uri() const
     return m_uri;
 }
 
+void DockPageView::open(const QVariantMap& params)
```

Comment:
What do you think about renaming this method to setParams?

Reply:
Good idea. That way, it's a bit more general, so it is more natural to reuse it for other things in the future.